src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    const glue = separator === null ? 'null' : separator;
+
+    for (let i = 0; i < this.length; i++) {
+      const currentString = this[i] === null || this[i] === undefined
+        ? ''
+        : this[i];
+
+      if (i === this.length - 1) {
+        return joinedString + currentString;
+      }
+
+      joinedString += currentString + glue;
+    }
```

Comment:
dont need to crate new variable(currentString), you can add current element to already created variable "joinedString", then check if it not last element and add separator.